src/main/java/beam/physsim/cch/RoutingFrameworkTravelTimeCalculator.scala

```diff
@@ -0,0 +1,230 @@
+package beam.physsim.cch
+
+import java.util
+import java.util.concurrent.Executors
+
+import beam.agentsim.events.PathTraversalEvent
+import beam.sim.BeamServices
+import com.google.common.util.concurrent.ThreadFactoryBuilder
+import com.typesafe.scalalogging.LazyLogging
+import com.vividsolutions.jts.geom.Coordinate
+import org.apache.commons.lang.time.StopWatch
+import org.matsim.api.core.v01.network.Link
+import org.matsim.core.controler.events.IterationEndsEvent
+
+import scala.collection.JavaConverters._
+import scala.collection.convert.ImplicitConversionsToScala._
+import scala.collection.mutable
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, Future}
+
+class RoutingFrameworkTravelTimeCalculator(
+  private val beamServices: BeamServices,
+  private val osmInfoHolder: OsmInfoHolder,
+  private val routingFrameworkWrapper: RoutingFrameworkWrapper,
+) extends LazyLogging {
+
+  private implicit val executionContext: ExecutionContext = ExecutionContext.fromExecutorService(
+    Executors.newFixedThreadPool(
+      Math.max(Runtime.getRuntime.availableProcessors() / 4, 1),
+      new ThreadFactoryBuilder().setDaemon(true).setNameFormat("routing-framework-worker-%d").build()
+    )
+  )
+
+  private val odsFactor: Int =
```

Comment:
Is it better to leave it as double, but do rounding to the nearest integer in the needed place? An example, let say I have 2500 OD pairs, odsFactor = 2.25. If I just use integer, the total number of OD pairs will be 2500 * 2 = 5000. But in case of double it will be 2500 * 2.25 = 5625

Reply:
In this case we need to change the logic, how it will work, because right now i am just writing each od **OD_FACTOR** number of times.

Reply:
fixed